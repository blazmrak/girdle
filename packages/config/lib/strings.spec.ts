import { initializeEnvironment, Var } from './index'
import { expectTypeOf } from 'expect-type'

describe('string', () => {
  it('parses basic options', () => {
    const env = { STR: 'str', OD: 'overriden' }

    const config = initializeEnvironment(
      {
        defaultStr: Var.string().default('default'),
        od: Var.string().default('default'),
        optionalStr: Var.string().optional(),
        str: Var.string(),
        namedStr: Var.string().name('STR'),
      },
      { env },
    )

    expect(config).toMatchSnapshot()
    expectTypeOf(config).toEqualTypeOf<{
      defaultStr: string
      od: string
      optionalStr: string | undefined
      str: string
      namedStr: string
    }>()
  })
})
