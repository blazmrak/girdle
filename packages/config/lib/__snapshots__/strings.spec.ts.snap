// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`string parses basic options 1`] = `
{
  "defaultStr": "default",
  "namedStr": "str",
  "od": "overriden",
  "optionalStr": undefined,
  "str": "str",
}
`;
